lib/teiserver/account/libs/relationship_lib.ex

```diff
@@ -154,6 +163,43 @@ defmodule Teiserver.Account.RelationshipLib do
     :ok
   end
 
+  @doc """
+  Checks if a user has reached the maximum number of relationships of a given type.
+
+  Returns `:ok` if the user is under the limit, or `{:error, reason}` if they've reached it.
+  The limit is configurable via the admin interface.
+  """
+  @spec check_relationship_limit(T.userid(), :friend | :ignore | :avoid | :block) ::
+          :ok | {:error, String.t()}
+  def check_relationship_limit(user_id, relationship_type) do
+    max_relationships =
+      Config.get_site_config_cache("relationships.Maximum relationships per user")
```

Comment:
To ease the testing bits, I suggest the `max_relationships` be a parameter with this default value